### PR TITLE
Feature/email client - stripping out and starting again

### PR DESCRIPTION
**Pull request by daniel-samson**
## Description

This change is the first step to creating the new email client.

# Goals
* use the existing views in SuiteCRM
* keep the views consistent across SuiteCRM

# Changes
- I have stripped out the old email client.
- I have added some new features to the list view.

Using the listviewdefs.php you can now:
* add buttons to the list view (in the same way you do for the edit view)
* include javascript to handle the buttons (in the same way you do for the edit view)
* hide the edit view link (in addition to changing it in the sugar view)
* hide the header label (used for the indicator fields)

Example of the listviewdefs.php:
```
$viewdefs['Emails']['ListView'] = array(
    'templateMeta' => array(
        'form' => array(
            'buttons' =>
                array(
                    array(
                        'customCode' => '<a class="btn" data-action="compose-email">{$MOD.LBL_COMPOSEEMAIL} <span class="glyphicon glyphicon-envelope"></span></a>'
                    ),
                ),
        ),
        'includes' => array(
            0 =>
                array(
                    'file' => 'modules/Emails/javascript/list.view.js',
                ),
        ),
        'options' => array(
            'hide_edit_link' => true
        )
    )
);
```
```
'STATUS'=> array(
        'width' => '32',
        'label' => 'LBL_LIST_STATUS',
        'default' => false,
        'hide_header_label' => false
    ),
```
I have added two function fields to the list view for emails. An indicator for new and flagged emails and an indicator for when the email has attachments.

## Motivation and Context

* Feature for 7.9

## How To Test This

* Rebuild Sass
* Repair and Rebuild
* Set up scheduler / cron
* Create a group email
* Check that you can access the list view when you navigate to emails
* Check that email templates is now treated as it's own module.

screenshot_20170303_094248

**Please note that this is a work in progress. Some of the functionality is missing.**

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

### Final checklist

- [x] My code follows the code style of this project found [here](https://suitecrm.com/wiki/index.php/Coding_Standards).
- [x] My change requires a change to the documentation.
- [X] I have read the [**How to Contribute**](https://suitecrm.com/wiki/index.php/Contributing_to_SuiteCRM) guidelines.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -75,35 +75,16 @@ public/
...
-themes/SuiteP/cases.css.map
-themes/SuiteP/campaigns.css
-themes/SuiteP/campaigns.css.map
-themes/SuiteP/calendar.css
-themes/SuiteP/calendar.css.map
-themes/SuiteP/admin.css
-themes/SuiteP/admin.css.map
\ No newline at end of file
+themes/SuiteP/css/*.map
+themes/SuiteP/css/*.css
```
Comment (gymad) [inline .gitignore]: we don't want to remove the themes/SuiteP/css/style.css from git repo?

File: include/ListView/ListViewButtons.tpl [hunk 2]
```diff
@@ -0,0 +1,13 @@
+{if !isset($templateMeta.form.buttons)}
+
```
Comment (gymad) [inline include/ListView/ListViewButtons.tpl]: why is the empty if..else statement?

File: themes/SuiteP/css/listview.scss [hunk 3]
```diff
@@ -3350,6 +3350,27 @@ ul.clickMenu > li > span.searchAppliedAlert {
...
+}
+
+.list-view-action-buttons > li {
+  display: inline-block;
+  background-color: transparent;
+  padding: 0 2px 0 2px;
+}
+
+.list-view-action-buttons > li .btn {
+  background-color: $list-view-action-button-bg;
```
Comment (gymad) [inline themes/SuiteP/css/listview.scss]: may better, if properties are in alphabetical order

File: themes/SuiteP/css/listview.scss [hunk 4]
```diff
@@ -3350,6 +3350,27 @@ ul.clickMenu > li > span.searchAppliedAlert {
...
 }
 
+// list-view-action-buttons
+
+.list-view-action-buttons {
+  margin-right: 5px;
+}
+
+.list-view-action-buttons > li {
+  display: inline-block;
```
Comment (gymad) [inline themes/SuiteP/css/listview.scss]: may better, if properties are in alphabetical order

File: modules/Emails/javascript/list.view.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+   alert('check new email placeholder');
+ });
+
+ $('[data-action=emails-open-folder]').click(function(){
+   alert('open folder email placeholder');
+ });
+
+ // look for new
+  $('.email-indicator .email-new').each(function(i, v){
+    $(this).parents('tr').addClass('email-new-record');
```
Comment (gymad) [inline modules/Emails/javascript/list.view.js]: add this class directly to .tpl (or use closest() instead of parents() at least)

**Comment by gymad** [comment]
If I remove the 'From Name' via colums filter the from addresses disapear
from_name

**Comment by daniel-samson** [comment]
Hi @Mausino. I appreciate the praise and I understand the how important the IMAP keywords are to you. 

This question should really be asked on the forums. This is not the correct place to add such a comment. Please only comment in relation to this pull request. For example if you find an issue with the changes I have made.

It is possible to filter / tag emails. Though it isn't a built in feature. You could use the workflow module to achieve a similar result.

File: themes/SuiteP/include/ListView/ListViewGeneric.tpl [hunk 6]
```diff
@@ -317,3 +324,15 @@ function lvg_nav(m,id,act,offset,t){
 </script>
 <script type="text/javascript" src="include/InlineEditing/inlineEditing.js"></script>
 {/if}
+
+
+{if isset($templateMeta)}
```
Comment (samus-aran) [inline themes/SuiteP/include/ListView/ListViewGeneric.tpl]: Whats the requirement for this?

Comment (daniel-samson) [inline themes/SuiteP/include/ListView/ListViewGeneric.tpl]: This allows a developer to add buttons to the list view in the same manner they can for the detail view and the edit view.

**Comment by daniel-samson** [comment]
I updated the licenses in this PR

File: .gitignore (hunk 1 above)
Comment (daniel-samson) [inline .gitignore]: Please refer to change " c4fcba - Adding style.css exception in .gitignore"

File: themes/SuiteP/include/ListView/ListViewGeneric.tpl (hunk 6 above)
Comment (samus-aran) [inline themes/SuiteP/include/ListView/ListViewGeneric.tpl]: Ok, understand that - but should it be consistent with other areas of the system.  For example EditView includes 'includes' via the EditView2.php

$this->th->ss->assign('includes', isset($this->defs['templateMeta']['includes']) ? $this->defs['templateMeta']['includes'] : null);

**Comment by samus-aran** [comment]
@gymad  - I wasn't able to replicate this issue - https://github.com/salesagility/SuiteCRM/pull/3215#issuecomment-284418030

Can you show me this?

**Comment by daniel-samson** [comment]
Updated PR so that the work the same way they do in the edit view.

**Pull request merged**